### PR TITLE
Small documentation change and remove hard coded domain setting

### PR DESCRIPTION
**Pull request by tom22222**

### CODE REVIEW

File: terraform/modules/puppetserver/templates/puppet.tpl [hunk 1]
```diff
@@ -25,7 +25,7 @@ runcmd:
   - echo -e "[main]\ncertname = ${fqdn}" >> /etc/puppetlabs/puppet/puppet.conf
   - sed -i '/\[main\]/a dns_alt_names=puppet.service.consul,puppetserver.service.consul,puppetmaster.service.consul' /etc/puppetlabs/puppet/puppet.conf
   - until /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.${domain}; do echo "puppet failed, retry in 10 seconds"; sleep 10; done
-  - until grep "@8600" /etc/unbound/unbound.conf; do echo "Consul still not configured, retry in 10 seconds"; sleep 10; /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.briggs.lan; done
+  - until grep "@8600" /etc/unbound/unbound.conf; do echo "Consul still not configured, retry in 10 seconds"; sleep 10; /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.${domain}; done
```
Comment (jaxxstorm) [inline terraform/modules/puppetserver/templates/puppet.tpl:28]: Probably needs to be `${::domain}` so it's name spaced correctly

File: terraform/modules/puppet_ca/templates/puppet.tpl [hunk 2]
```diff
@@ -31,7 +31,7 @@ runcmd:
   - r10k deploy environment -p --verbose -c /etc/puppetlabs/r10k.yaml
   - echo -e "nameserver 173.245.58.51\nnameserver 8.8.8.8" > /etc/resolv.conf
   - until /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.${domain}; do echo "puppet failed, retry in 10 seconds"; sleep 10; done
-  - until grep "@8600" /etc/unbound/unbound.conf; do echo "Consul still not configured, retry in 10 seconds"; sleep 10; /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.briggs.lan; done
+  - until grep "@8600" /etc/unbound/unbound.conf; do echo "Consul still not configured, retry in 10 seconds"; sleep 10; /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.${domain}; done
```
Comment (jaxxstorm) [inline terraform/modules/puppet_ca/templates/puppet.tpl:34]: Probably needs to be `${::domain}` so it's name spaced correctly

File: terraform/modules/consulserver/templates/consulserver.tpl [hunk 3]
```diff
@@ -23,7 +23,7 @@ runcmd:
   - echo -e "nameserver 173.245.58.51\nnameserver 8.8.8.8" > /etc/resolv.conf
   - echo -e "[main]\ncertname = ${fqdn}" >> /etc/puppetlabs/puppet/puppet.conf
   - until /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.${domain}; do echo "puppet failed, retry in 10 seconds"; sleep 10; done
-  - until grep "@8600" /etc/unbound/unbound.conf; do echo "Consul still not configured, retry in 10 seconds"; sleep 10; /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.briggs.lan; done
+  - until grep "@8600" /etc/unbound/unbound.conf; do echo "Consul still not configured, retry in 10 seconds"; sleep 10; /opt/puppetlabs/puppet/bin/puppet agent -t --environment=develop --server=puppetserver-0.${domain}; done
```
Comment (jaxxstorm) [inline terraform/modules/consulserver/templates/consulserver.tpl:26]: Probably needs to be `${::domain}` so it's name spaced correctly

**Review by jaxxstorm — CHANGES_REQUESTED** [review]
Just a small tweak to update the namespace of the facter var, but other than thanks for the contribution!

File: terraform/modules/puppetserver/templates/puppet.tpl (hunk 1 above)
Comment (tom22222) [reply terraform/modules/puppetserver/templates/puppet.tpl:28]: I'm not an expert on this stuff but isn't this still a terraform variable at this stage (see the previous line of code that uses ${domain}, is it not the same as that?)

File: terraform/modules/puppet_ca/templates/puppet.tpl (hunk 2 above)
Comment (tom22222) [reply terraform/modules/puppet_ca/templates/puppet.tpl:34]: I'm not an expert on this stuff but isn't this still a terraform variable at this stage (see the previous line of code that uses ${domain}, is it not the same as that?)

File: terraform/modules/consulserver/templates/consulserver.tpl (hunk 3 above)
Comment (tom22222) [reply terraform/modules/consulserver/templates/consulserver.tpl:26]: I'm not an expert on this stuff but isn't this still a terraform variable at this stage (see the previous line of code that uses ${domain}, is it not the same as that?)

**Comment by tom22222** [comment]
thanks for getting back, pls see my comments on the proposed changes

**Review by jaxxstorm — APPROVED** [review]

**Comment by jaxxstorm** [comment]
You're right, didn't check it properly! Thanks for the contribution

**Pull request merged**